node/network/collator-protocol/src/collator_side/mod.rs

```diff
@@ -698,6 +765,91 @@ where
 				);
 			}
 		},
+		Preconnect(relay_parent) => {
```

Comment:
Will it be fine to take any head from the view and use it for the session & group rotation info request?

Reply:
We should make sure to pre-connect for all active heads, given that group rotation happen based on block numbers and forks might have different heights, this will be necessary.